### PR TITLE
CORE-461 (This could work)

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: CpEpEngine.cpp [hunk 1]
```diff
@@ -2066,14 +2061,14 @@ STDMETHODIMP CpEpEngine::ConfigPassphraseForNewKeysByEmail(VARIANT_BOOL enable,
...
-    PEP_STATUS status = ::config_passphrase_for_new_keys_by_email(session(), enable, _accountEmail.c_str(), _passphrase.c_str());
-
-    if (status == PEP_STATUS_OK) {
-        return S_OK;
+    if (enable) {
+        passphrase_cache.add(_accountEmail, _passphrase);
     }
     else {
-        return FAIL(L"ConfigPassphraseForNewKeysByEmail", status);
+        passphrase_cache.remove(_accountEmail, _passphrase);
```
Comment (ig-garcia) [inline CpEpEngine.cpp]: Same as in libPlanckWrapper, should we not remove just by email?

Comment (ghost) [reply CpEpEngine.cpp]: I've made the implementation and the interface more explicit. Please check again.

**Review by saschabacardit — APPROVED** [review]

**Review by ip-plank-security — APPROVED** [review]

File: CpEpEngine.cpp [hunk 2]
```diff
@@ -2595,28 +2591,27 @@ STDMETHODIMP CpEpEngine::SignatureVerifies(BSTR text, BSTR signature, VARIANT_BO
...
     const PEP_STATUS status = manage_passphrase(session(), account_passphrases, passphrase.c_str(), &_error_accounts);
 
-    free_stringpair_list(account_passphrases);
-
     if (status != PEP_STATUS_OK && _error_accounts) {
         *error_accounts = string_array(_error_accounts);
     }
     else if (status == PEP_STATUS_OK) {
-        // cache the new passphrase
-        cache_passphrase(passphrase);
```
Comment (ig-garcia) [inline CpEpEngine.cpp:2617]: I see we don't cache passphrase without email anymore, I will update that and verify behavior.

**Review by ig-garcia — APPROVED** [review]

File: CpEpEngine.cpp (hunk 2 above)
Comment (ghost) [reply CpEpEngine.cpp:2617]: At least not in this part of the API. Is still supported though by the passphrase cache, and can still get triggered. E.g. by "losing a key sync".

**Pull request merged**